clients/js/src/create-mint.ts

```diff
@@ -0,0 +1,131 @@
+  unwrappedMint: Address;
+  wrappedTokenProgram: Address;
+  payer: KeyPairSigner;
+  idempotent: boolean;
+}) => {
+  const [wrappedMint] = await findWrappedMintPda(
+    {
+      unwrappedMint,
+      wrappedTokenProgram: wrappedTokenProgram,
+    },
+    { programAddress: TOKEN_WRAP_PROGRAM_ADDRESS },
+  );
+  const [backpointer] = await findBackpointerPda(
+    { wrappedMint },
+    { programAddress: TOKEN_WRAP_PROGRAM_ADDRESS },
+  );
+
+  const instructions: IInstruction[] = [];
+
+  // Fund wrapped mint account if needed
+  let fundedWrappedMintLamports = 0n;
+  const wrappedMintAccount = await rpc.getAccountInfo(wrappedMint).send();
+  const mintSize = BigInt(getMintSize());
+  const wrappedMintRent = await rpc.getMinimumBalanceForRentExemption(mintSize).send();
+  const wrappedMintLamports = wrappedMintAccount.value?.lamports ?? 0n;
+  if (wrappedMintLamports < wrappedMintRent) {
+    fundedWrappedMintLamports = wrappedMintRent - wrappedMintLamports;
+    instructions.push(
+      getTransferSolInstruction({
+        source: payer,
+        destination: wrappedMint,
+        amount: fundedWrappedMintLamports,
+      }),
+    );
+  }
+
+  // Fund backpointer account if needed
+  let fundedBackpointerLamports = 0n;
+  const backpointerAccount = await rpc.getAccountInfo(backpointer).send();
+  const backpointerSize = BigInt(getBackpointerSize());
+  const backpointerRent = await rpc.getMinimumBalanceForRentExemption(backpointerSize).send();
+  const backpointerLamports = backpointerAccount.value?.lamports ?? 0n;
+  if (backpointerLamports < backpointerRent) {
+    fundedBackpointerLamports = backpointerRent - backpointerLamports;
+    instructions.push(
+      getTransferSolInstruction({
+        source: payer,
+        destination: backpointer,
+        amount: fundedBackpointerLamports,
+      }),
+    );
+  }
+
+  // Add create_mint instruction
+  instructions.push(
+    getCreateMintInstruction({
+      wrappedMint,
+      backpointer,
+      unwrappedMint,
+      wrappedTokenProgram,
+      idempotent,
+    }),
+  );
+
+  const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
+
+  // Build transaction
+  let tx = pipe(
+    createTransactionMessage({ version: 0 }),
+    tx => setTransactionMessageFeePayer(payer.address, tx),
+    tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, tx),
+  );
+
+  for (const instruction of instructions) {
+    tx = appendTransactionMessageInstruction(instruction, tx);
+  }
+
+  const compiledTransaction = compileTransaction(tx);
+
+  // Sign and send
+  const [signatures] = await payer.signTransactions([compiledTransaction]);
+  if (!signatures) {
+    throw new Error('Expected a signature for compiled transaction');
+  }
+
+  const wireFormatTransaction = getBase64EncodedWireTransaction({
+    ...compiledTransaction,
+    signatures,
+  });
+
+  const signature = await rpc.sendTransaction(wireFormatTransaction, { encoding: 'base64' }).send();
```

Comment:
All this can be replaced with:

```suggestion
   // You passed signers everywhere so we know who needs to sign the transaction and how.
  const signedTransaction = await signTransactionMessageWithSigners(tx);
  
  // Create a send and confirm function from your RPC and RPC Subscriptions objects.
  // You can create the latter using `createSolanaRpcSubscriptions`.
  const sendAndConfirm = sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions });
 
  // Use it to send and confirm your signed transaction.
  await sendAndConfirm(signedTransaction, { commitment: 'confirmed' });
  
  // Access the signature of your signed transaction (this can be done even before sending it).
  const signature = getSignatureFromTransaction(signedTransaction);
```